From 598f3b4fdbbc2f1d5b544719038e447191d792ad Mon Sep 17 00:00:00 2001
From: Henry Wang <Henry.Wang@arm.com>
Date: Thu, 21 Jul 2022 04:56:22 +0000
Subject: [PATCH] xen/arm: Introduce a Kconfig to enable MPAM

This commit introduces a Kconfig `CONFIG_MPAM` for users to enable
MPAM if needed. Currently, this Kconfig depends on `CONFIG_ARM64`
since all currently-known models that have MPAM extension are 64bits.

The default value of `CONFIG_MPAM` is set to n.

Signed-off-by: Henry Wang <Henry.Wang@arm.com>
Upstream-Status: Pending
Issue-Id: SCM-5049
---
 xen/arch/arm/Kconfig  | 10 ++++++++++
 xen/arch/arm/Makefile |  1 +
 2 files changed, 11 insertions(+)

diff --git a/xen/arch/arm/Kconfig b/xen/arch/arm/Kconfig
index 2939db429b78..0c1f1f3c1884 100644
--- a/xen/arch/arm/Kconfig
+++ b/xen/arch/arm/Kconfig
@@ -182,6 +182,16 @@ config ARM64_BTI
 	  Branch Target Identification support.
 	  This feature is not supported in Xen.
 
+config MPAM
+	bool "MPAM (Memory System Resource Partitioning and Monitoring) Support (UNSUPPORTED)" if UNSUPPORTED
+	def_bool n
+	depends on ARM_64
+	help
+	  This option enables the Xen support for Arm Memory System Resource
+	  Partitioning and Monitoring (MPAM), which is an optional architecture
+	  feature on post Armv8.4 system that describes an interface to control
+	  cache and memory bandwidth.
+
 source "arch/arm/tee/Kconfig"
 
 config STATIC_SHM
diff --git a/xen/arch/arm/Makefile b/xen/arch/arm/Makefile
index 7bf07e992046..c5c8639e66cb 100644
--- a/xen/arch/arm/Makefile
+++ b/xen/arch/arm/Makefile
@@ -38,6 +38,7 @@ obj-$(CONFIG_LIVEPATCH) += livepatch.o
 obj-y += mem_access.o
 obj-y += mm.o
 obj-y += monitor.o
+obj-$(CONFIG_MPAM) += mpam.o
 obj-y += p2m.o
 obj-y += percpu.o
 obj-y += platform.o
-- 
2.34.1

